### PR TITLE
[DOC beta] fix route import in routeWillChange/routeDidChange examples

### PR DESCRIPTION
**Pull request by knownasilya**

### CODE REVIEW

File: packages/@ember/-internals/routing/lib/services/router.ts [hunk 1]
```diff
@@ -412,19 +412,19 @@ export default class RouterService extends Service {
...
+    export default class ContactFormRoute extends Route {
       @service router;
 
       constructor() {
         super(...arguments);
 
         this.router.on('routeWillChange', (transition) => {
           if (!transition.to.find(route => route.name === this.routeName)) {
-            alert("Please save or cancel your changes.");
+            alert('Please save or cancel your changes.');
```
Comment (knownasilya) [inline packages/@ember/-internals/routing/lib/services/router.ts:427]: Should this be a `confirm` in the if expression?

Comment (snewcomer) [reply packages/@ember/-internals/routing/lib/services/router.ts:427]: confirm seems right!

**Review by snewcomer — APPROVED** [review]
Nice improvements!

File: packages/@ember/-internals/routing/lib/services/router.ts [hunk 2]
```diff
@@ -412,19 +412,20 @@ export default class RouterService extends Service {
...
 
       constructor() {
         super(...arguments);
 
         this.router.on('routeWillChange', (transition) => {
-          if (!transition.to.find(route => route.name === this.routeName)) {
-            alert("Please save or cancel your changes.");
+          const isTransitioningAway = !transition.to.find(route => route.name === this.routeName);
+          
+          if (isTransitioningAway && confirm('Are you sure you want to continue? Please save or cancel your changes first.')) {
```
Comment (MelSumner) [inline packages/@ember/-internals/routing/lib/services/router.ts:428]:
```suggestion
          const isTransitioningAway = !transition.to.find(route => route.name === this.routeName);

          if (isTransitioningAway && confirm('Are you sure you want to continue? Please save or cancel your changes first.')) {
```

**Comment by knownasilya** [comment]
Would love to, but away from laptop for next few days.

**Comment by kategengler** [comment]
@knownasilya Can you retarget to master?

**Comment by knownasilya** [comment]
Will try to rebase later tomorrow. Should be pointing to master now.

**Comment by chriskrycho** [comment]
@knownasilya is this still needed, and if so, can you rebase? I'll see about getting it landed. (11 months later is not too late if it's still wrong in our docs! 😁)

**Comment by kategengler** [comment]
Is this still relevant? It is a draft PR

**Pull request closed without merging**